### PR TITLE
Add GOCACHE in build.sh

### LINKED ISSUE #108: Cache build results across xgo runs
xgo compilation is very slow because it uses `go build` internally, [which doesn't cache cgo compilation results](https://stackoverflow.com/questions/32061750/go-build-rebuilds-unnecessarily).

Maybe we can mount ~/.xgo-cache/<platform> as `GOCACHE` directory within the container, which will be reused across xgo runs?

### PR DESCRIPTION
**Pull request by leventov**
Fixes #108

### CODE REVIEW

File: xgo.go [hunk 1]
```diff
@@ -315,11 +315,17 @@ func compile(image string, config *ConfigFlags, flags *BuildFlags, folder string
 	re := regexp.MustCompile("([A-Z]):")
 	folder_w := filepath.ToSlash(re.ReplaceAllString(folder, "/$1"))
 	depsCache_w := filepath.ToSlash(re.ReplaceAllString(depsCache, "/$1"))
+	gocache := filepath.Join(depsCache, "gocache")
+	if err := os.MkdirAll(gocache, 0777); err != nil {
```
Comment (kolaente) [inline xgo.go:319]: Why the `777`?

Comment (kolaente) [reply xgo.go:319]: I think 644 should be fine

Comment (leventov) [reply xgo.go]: Thanks, changed

**Review by kolaente — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
🤔  hmmm... CI fail is kinda related, I wonder if it is due to base not being rebuilt.

**Comment by leventov** [comment]
@techknowlogick what do you mean by "base not being rebuilt"?

**Comment by techknowlogick** [comment]
@leventov the base docker image, however I've toggled it on to build fresh for this PR and it still fails in same way, so that sadly wasn't the issue. Now I am rather confused as in theory that directory should be writeable, I wonder if it fails because it is created inside the docker container with root permissions and perhaps outside the docker container the CI runs with non-root user

**Comment by bearsh** [comment]
the problems seems to be that the `.xgo-cache` folder in the home directory is created from inside the container as root.
```bash
$ ls -la . | grep .xgo
drwxr-xr-x   2 root   root        4096 Jun 15  2020 .xgo-cache
```
 so it's not writable by this code:
```go
if err := os.MkdirAll(gocache, 0641); err != nil {
	log.Fatalf("Failed to create gocache dir: %v.", err)
}
```
I think `.xgo-cache` should be owned by the user and ideally also I files inside it. so xgo should create that folder initially (and error if it's already present with wrong ownership with a message how to fix it). hopefully it's working when the 
build script calls  `chown -R --reference /gochache /gocache/*` at the end like it does already on the `/build` folder.

File: xgo.go [hunk 2]
```diff
@@ -315,11 +315,17 @@ func compile(image string, config *ConfigFlags, flags *BuildFlags, folder string
 	re := regexp.MustCompile("([A-Z]):")
 	folder_w := filepath.ToSlash(re.ReplaceAllString(folder, "/$1"))
 	depsCache_w := filepath.ToSlash(re.ReplaceAllString(depsCache, "/$1"))
+	gocache := filepath.Join(depsCache, "gocache")
+	if err := os.MkdirAll(gocache, 0641); err != nil {
```
Comment (bearsh) [inline xgo.go:319]: is this `0641` intended?

Comment (leventov) [reply xgo.go:319]: It was an attempt to shrink down from 644 suggested above: https://github.com/techknowlogick/xgo/pull/116#discussion_r629225549

**Comment by leventov** [comment]
@bearsh please feel free to recreate or fix this PR if you understand how to fix it.

File: xgo.go (hunk 2 above)
Comment (zeripath) [reply xgo.go:319]: This is supposed to be a directory 0644 = rw-r--r-- which means you can't actually traverse the directory

This should be 0755 or 0750

Comment (zeripath) [reply xgo.go:319]: I'm gonna push up an update

**Comment by zeripath** [comment]
I think I've fixed this now.

**Comment by zeripath** [comment]
Refixed conflicts

**Comment by zeripath** [comment]
I'll rebase this once #194 is in.

**Comment by zeripath** [comment]
Conflicts re-resolved

**Pull request merged**